src/docs/arch-schema-ops.dox

```diff
@@ -0,0 +1,206 @@
+object on the filesystem with the right parameters and then creating an entry
+for this new object into the metadata. The sequence of operations involved in a
+create for various schema types are as follows:
+
+@plantuml_start{schema_create.png}
+@startuml{schema_create.png}
+:WT_SESSION->create(.,name,.)
+(__session_create());
+
+partition session-API-call {
+    :API session init with NULL dhandle;
+    :exit if PANIC flag set;
+    :exit if invalid configuration;
+}
+
+:validate "name" and if passed "type" config parameter;
+note right
+    "name" parameter is called as "uri" internally.
+    "type" is usually not passed and generally
+    implied from the uri.
+end note
+
+partition with-schema-lock {
+    partition with-table-lock {
+        :turn on meta tracking;
+        :check uri}
+
+        split
+            :uri matches "file:"
+            ("file" is the underlying
+             type for all the objects);
+        split again
+            :uri matches "colgroup:";
+            :__create_colgroup();
+        split again
+            :uri matches "table:";
+            :__create_table();
+        split again
+            :uri matches "lsm:";
+            :__wt_lsm_tree_create();
+        split again
+            :uri matches "index:";
+            :__create_index();
+        split again
+            :matches a named data-source;
+            :__create_data_source();
+        end split
+
+        partition __create_file() {
+            :exit if file exists;
+            :validate allocation size;
+            :block manager creates the file:
+             1.create file using __wt_open()
+             2.write an initial descriptor to file
+             3.fsync and close the file handle;
+            if (metadata-file?) then (yes)
+            else (no)
+                :update metadata with file
+                 configuration and version;
+            endif
+            :check if file setup correctly by
+             getting btree handle with
+             WT_DHANDLE_EXCLUSIVE set;
+            if (metatracking on?) then (yes)
+                :track locked handle*;
+            else (no)
+                :release btree -
+                 sync and close;
+            endif
+        }
+
+        partition turn-off-meta-tracking {
+            if (errors?) then (yes)
+                :unroll operations;
+            else (no)
+                if (logging?) then (yes)
+                    :sync log;
+                else (no)
+                endif
+                :checkpoint and sync;
+            endif
+            :apply post-commit ops:
+             release tracked (handle) btree* -
+             sync and close;
+            note right
+                if meta tracking is on, this btree
+                was being tracked as locked. As part
+                of tuning off meta tracking, we sync
+                and close this btree
+            end note
+        }
+    }
+}
```

Comment:
I am kind of concerned about keeping this pseudo-code up to date with the actual code. This is pretty detailed. Is there a plan for that?

I see later that this already exists and this change is just a code movement.

Reply:
Right, the 'Schema Operations' section existed previously.  I believe I did a `git mv`, but somehow that didn't show up in the diff.